StepperEngine/src/Steps/CommandLineStep.java

```diff
@@ -46,11 +47,19 @@ protected void outerRunStepFlow(){
 
     @Override
     protected void runStepFlow() throws Exception {
+        String commandLineWithArguments = "";
         String commandLine=this.command.getData();
         String arguments=this.arguments.getData();
         addLog("About to invoke "+ commandLine + arguments);
         if(commandLine==null||commandLine.isEmpty())throw new Exception("empty command entered");
-        String commandLineWithArguments="cmd.exe /c "+commandLine+" "+arguments;
+
+        if(System.getProperty("os.name").toLowerCase().contains("windows"))
```

Comment:
This is not what I meant but as long as it works it's fine by me

Java is suppose to be multi-platform, so a code like this seems weird to me. I'd think Java should be able to determine the default command line tool on it's own, at run time.  Did you try not to speficy it at all and see what happens?

Reply:
i looked around stackoverflow, and for each different os they suggested to add a specific string for the corresponding command line, maybe there is an object that can generify it all, but the instructor wants us to use ProcessBuilder which does not .